ubermag/basic_logging.py

```diff
@@ -0,0 +1,58 @@
+"""Setup basic logging for all ubermag packages."""
+import logging
+
+
+def setup_logging(level=logging.WARNING, package_levels=None):
+    """Set up basic logging for Ubermag with per-package control.
+
+    This function creates a basic logger (printing to stdout) and sets the log
+    level for all packages in Ubermag to ``level``. Additional, more fine-grain
+    control is possible by passing a dictionary to ``package_levels``. Keys
+    must be ubermag subpackages, values log levels.
+
+    Parameters
+    ----------
+    level : str, int, logging.LEVEL
+        Log level used for all packages in Ubermag.
+
+    package_levels : dict, optional
+        Dictionary with Ubermag subpackage names as keys and log-levels
+        as values. It allows fine-grain control over logging for
+        individual packages.
+
+    Example
+    -------
+    1. Setting up a basic logger with default log-level ``logging.WARNING``
+
+    >>> import ubermag
+    >>> ubermag.setup_logging()
+
+    2. Setting up a basic logger with a higher log-level for ``oommfc``
+
+    >>> import ubermag
+    >>> ubermag.setup_logging(package_levels={'oommfc': logging.DEBUG})
+
+    """
+    # TODO Each package should use a single logger.
+    packages = [
+        'discretisedfield',
+        'mag2exp',
+        'micromagneticdata',
+        'micromagneticmodel',
+        'micromagnetictests',
+        'oommfc',
+        'ubermagtable',
+        'ubermagutil',
+    ]
+
+    # TODO Use better configuration, e.g.:
+    # <LEVEL>: <package>/<file>/line <line-number>: <msg>
```

Comment:
Here is a suggestion for potential output formatting (together with example program for easier testing):

```python
#!/usr/bin/env python
import logging
import time


# one possible output
logging.basicConfig(format='%(asctime)s,%(msecs)d %(levelname)-8s [%(filename)10s:%(funcName)s:%(lineno)d] %(message)s',
                    datefmt='%Y-%m-%d:%H:%M:%S', level=logging.DEBUG)
logger = logging.getLogger(__name__)


def func1():
    logger.debug("This is a debug log")
    logger.info("This is an info log")


def func2():
    logger.warning("Warning entry")
    logger.error("And an error")
    logger.critical("Critical entry")


func1()
time.sleep(1)
func2()
time.sleep(1)
func1()


```

What I haven't tested yet is how this would look like with logging messages coming from different python packages (I.e. will it be clear from which package the message is - maybe the file name is not sufficiently unique (thinking of `__init__.py` or `lib.py` or so).

Other potentially useful logger format options are `%(pathname)s` (Full pathname of the source file where the logging call was issued(if available).) and `%(module)s` (Module (name portion of filename)).

If we think the output is long, we could shorten the datetime (`20210813T170101` instead of `2021-08-13:17:01:01`), and/or get rid of the milliseconds.